sui_programmability/examples/games/sources/frenemies.move

```diff
@@ -0,0 +1,387 @@
+module games::frenemies {
+    use std::vector;
+    use sui::address;
+    use sui::event;
+    use sui::math;
+    use sui::object::{Self, ID, UID};
+    use sui::sui_system::{Self, SuiSystemState};
+    use sui::table::{Self, Table};
+    use sui::transfer;
+    use sui::tx_context::{Self, TxContext};
+    use sui::validator_set;
+    use sui::validator;
+
+    #[test_only]
+    use sui::test_scenario::{Self as ts, Scenario};
+
+    /// Singleton struct scoring staking infomration for recent epochs,
+    /// start epoch, and a leaderborad with top scorers
+    struct Leaderboard has key {
```

Comment:
Not sure if "Leaderboard" is a good name. A leaderboard should contain players with highest scores, who are, in our case, players who have completed their goals and with highest difficulties. This doesn't really have that info right now.

Reply:
Good point. I started off thinking about keeping the score in here, but wasn't sure whether a "top 100 or so" in a single field impl, a dynamic fields-based impl, or events-based leaderboard impl would be best from a frontend perspective.

I'll go ahead with a simple "top 500 in a single field" approach for now, but we can tweak this if needed.